tests/device_tests/test_protection_levels.py

```diff
@@ -32,239 +33,362 @@
 PIN4 = "1234"
 
 
-@pytest.mark.skip_t2
-class TestProtectionLevels:
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_initialize(self, client):
-        with client:
-            client.set_expected_responses([messages.Features])
-            client.init_device()
-
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_apply_settings(self, client):
-        with client:
-            client.use_pin_sequence([PIN4])
-            client.set_expected_responses(
-                [
-                    messages.PinMatrixRequest,
-                    messages.ButtonRequest,
-                    messages.Success,
-                    messages.Features,
-                ]
-            )  # TrezorClient reinitializes device
-            device.apply_settings(client, label="nazdar")
-
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_change_pin(self, client):
-        with client:
+pytestmark = pytest.mark.setup_client(pin=PIN4, passphrase=True)
+
+# General NOTE: we need to assert the value of client.features.passphrase_protection
+# at the end, because in case of TT, this value gets filled correcly only after
+# unlocking the device with the PIN.
+# Also, it may be needed to update the client info by client.refresh_features().
+# There are cases when it is even impossible to check the correct TT passphrase_protection,
```

Comment:
if the test doesn't unlock by itself, please use `client.ensure_unlocked()`. you might need to specify a pin sequence for the unlock.

maybe an overall better solution would be to create something like:
```
def _assert_protection_enabled(client):
	with client:
		client.use_pin_sequence(...)
		client.ensure_unlocked()
		assert pin_protection is True
		assert passphrase_protection is True
	client.clear_session()
```
and then call it at start of every test?

Reply:
Tests that do not have both can have the assertions listed explicitly. If PIN is disabled, TT will correctly report `passphrase_protection` status at start.

If PIN is enabled and passphrase is not, we should probably try to unlock and check if passphrase protection is False? So maybe this function should have an argument, `passphrase_expected=True` that you can explicitly specify as False?

Reply:
Fixed in 699ae28